npm/pkg/dataplane/dataplane_linux.go

```diff
@@ -37,3 +53,81 @@ func (dp *DataPlane) refreshPodEndpoints() error {
 	// NOOP in Linux
 	return nil
 }
+
+// detectIptablesVersion sets the global iptables variable to nft if detected or legacy if detected.
+// NPM will crash if it fails to detect either.
+// This global variable is referenced in all iptables related functions.
+func detectIptablesVersion(ioShim *common.IOShim) error {
+	klog.Info("first attempt detecting iptables version. running: iptables-nft-save -t mangle")
+	cmd := ioShim.Exec.Command(util.IptablesSaveNft, "-t", "mangle")
+	output, err := cmd.CombinedOutput()
+	if err == nil && strings.Contains(string(output), "KUBE-IPTABLES-HINT") || strings.Contains(string(output), "KUBE-KUBELET-CANARY") {
```

Comment:
You should almost always check `err != nil` first, since the convention is that if that is true, `output` is nonsense.

Reply:
I agree, although in this case the current check seems better to me since the function returns only if this check is true

Reply:
What happens if `iptables-nft-save` isn't in $PATH though?

Reply:
Resolving since this logic is quite outdated. Default is nft now